### PR TITLE
Making a fallback for broken auth on github

### PR DESCRIPTION
**Pull request by matthiaz**
TLDR: We shouldn't be adding authentication on anything but the api.github.com url

Following the discussion about github deprecating auth via query string https://github.com/composer/composer/issues/8454
(Original blog post here https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters )

A fix was created see: https://github.com/composer/composer/commit/4b6c25d4bc33d49097320e29e6e5705b12e9d6ef

The problem is that this broke hardcoded URLs to tag archives. This works:

```
% curl -sv https://github.com/components/codemirror/archive/5.48.2.zip |& grep HTTP
> GET /components/codemirror/archive/5.48.2.zip HTTP/1.1
< HTTP/1.1 302 Found
```

Add a scopeless token, and it doesn't redirect anymore:

```
% curl -sv -H "Authorization: token <TOKENHERE>" https://github.com/components/codemirror/archive/5.48.2.zip |& grep HTTP
> GET /components/codemirror/archive/5.48.2.zip HTTP/1.1
< HTTP/1.1 404 Not Found
```

Why is this a problem?

In the PHP world, it is really not uncommon to see composer.lock files that have hardcoded URLs like this. There are a number of them hosted publicly on GitHub. [Here is a random example](https://github.com/eGovPDX/portlandor/blob/master/composer.lock):

```
        {
            "name": "codemirror/codemirror",
            "version": "5.48.2",
            "dist": {
                "type": "zip",
                "url": "https://github.com/components/codemirror/archive/5.48.2.zip"
            },
            "require": {
                "composer/installers": "~1.0"
            },
            "type": "drupal-library",
            "extra": {
                "installer-name": "codemirror"
            }
        },
```

This used to work, now it doesn't.

### CODE REVIEW

**Comment by damz** [comment]
Alternatively, we could only add the authentication token when the domain is literally `https://api.github.com`. Adding authentication on `https://github.com` itself never worked, and was only harmless because it was just adding a query parameter.

**Comment by matthiaz** [comment]
@damz Thanks for the pointer, I've changed the PR to reflect this.

**Comment by damz** [comment]
:+1: That sounds right to me. I think we could even restrict this to only `https`, because sending authentication credentials over unencrypted HTTP sounds like a bad idea to begin with.

**Comment by Seldaek** [comment]
Thanks, I think this makes a lot of sense, funny how things can be wrong yet working for ages sometimes.
